### PR TITLE
Sprint 31 TLT-1949 Use lti_permissions framework

### PR DESCRIPTION
**Pull request by douglashall**
Check lti_permissions to authorize a user for viewing the tool and sending email to a Teaching Staff Only mailing list.

### CODE REVIEW

**Comment by sapnamysore** [comment]
@sapnamysore

File: lti_emailer/canvas_api_client.py [hunk 1]
```diff
@@ -95,12 +96,11 @@ def get_sections(canvas_course_id):
 
 
 def get_teaching_staff_enrollments(canvas_course_id):
-    account_id = canvas_api_helper_courses.get_course(canvas_course_id)['account_id']
     users = get_users_in_course(canvas_course_id)
     enrollments = []
     for user in users:
         for enrollment in user['enrollments']:
-            if enrollment['type'] in _get_authorized_teaching_staff_enrollment_types(account_id):
+            if is_allowed([enrollment['role']], 'lti_emailer_send_all', canvas_course_id):
```
Comment (sapnamysore) [inline lti_emailer/canvas_api_client.py]: Can you make  the permission'lti_emailer_send_all' a  constant?

File: mailing_list/api.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 from icommons_common.auth.lti_decorators import has_course_permission
```
Comment (sapnamysore) [inline mailing_list/api.py]: Can this import go away?

File: mailing_list/api.py [hunk 3]
```diff
@@ -22,7 +23,7 @@
 
 @login_required
 @lti_role_required(const.TEACHING_STAFF_ROLES)
-@has_course_permission(canvas_api_helper_courses.COURSE_PERMISSION_SEND_MESSAGES_ALL)
+@lti_permission_required('lti_emailer_view')
```
Comment (sapnamysore) [inline mailing_list/api.py]: Make 'lti_emailer_view' a contsant

File: mailing_list/api.py [hunk 4]
```diff
@@ -53,7 +54,7 @@ def lists(request):
...
```
Comment (sapnamysore) [inline mailing_list/api.py]: Same as above

File: mailing_list/views.py [hunk 5]
```diff
@@ -18,7 +20,7 @@
 
 @login_required
 @lti_role_required(const.TEACHING_STAFF_ROLES)
-@has_course_permission(canvas_api_helper_courses.COURSE_PERMISSION_SEND_MESSAGES_ALL)
+@lti_permission_required('lti_emailer_view')
```
Comment (sapnamysore) [inline mailing_list/views.py]: Same as above: make the permission a const in all the places

**Comment by sapnamysore** [comment]
Other than making the permissions constant for reusability and easy maintenance, looks good.

File: mailing_list/api.py (hunk 2 above)
Comment (douglashall) [inline mailing_list/api.py]: Yes, thanks.

**Comment by elliottyates** [comment]
Just curious, does this speed up load time?

**Comment by douglashall** [comment]
I have not done any profiling, but it should be significantly faster on the first load for some users and slightly slower on subsequent calls.

**Pull request merged**